src/broadcast/mpris/mpris.cpp

```diff
@@ -0,0 +1,58 @@
+#include "broadcast/mpris/mpris.h"
+
+#include <QtDBus/QtDBus>
+
+#include "broadcast/mpris/mediaplayer2.h"
+#include "broadcast/mpris/mediaplayer2player.h"
+#include "broadcast/mpris/mediaplayer2playlists.h"
+#include "broadcast/mpris/mediaplayer2tracklist.h"
+#include "moc_mpris.cpp"
+#include "track/track.h"
+
+namespace {
+const QString busName = "org.mpris.MediaPlayer2.mixxx";
+}
+
+Mpris::Mpris(MixxxMainWindow* pMixxx,
+        PlayerManagerInterface* pPlayerManager,
+        UserSettingsPointer pSettings)
+        : m_busConnection(QDBusConnection::connectToBus(QDBusConnection::SessionBus, busName)),
+          m_pPlayer(new MediaPlayer2Player(pPlayerManager,
+                  this,
+                  pSettings)) {
+    // Classes derived from QDBusAbstractAdaptor must be created
+    // on the heap using the new operator and must not be deleted
+    // by the user (they will be deleted automatically when the object
+    // they are connected to is also deleted).
+    // http://doc.qt.io/qt-5/qdbusabstractadaptor.html
+    new MediaPlayer2(pMixxx, this);
+    m_busConnection.registerObject("/org/mpris/MediaPlayer2", this);
```

Comment:
QLatin1String?

Reply:
IIRC QLatin1String can be used when comparing a QString to a literal value. When constructing an actual QString, I think QStringLiteral should be used.